app/Http/Controllers/Sponsors.php

```diff
@@ -152,6 +156,64 @@ private function getSponsors() {
         }
     }
 
+    private function getSponsorDisplay() {
+        // I also add on the SponsorDetail that includes the logo URL + description.
+        if(Auth::check() && in_array(Auth::user()->type, ["admin", "committee","sponsor","hacker"])) {
+            $sponsors = DB::table('sponsors')
+                ->leftJoin('sponsor_details',function($join) {
+                    $desired_type = "portal-info";
+                    $join->on('sponsors.id','=','sponsor_details.sponsor_id')
+                        ->where('sponsor_details.type','=',$desired_type);
+                })
```

Comment:
I think you are overcomplicating the `join` too much.

I will use something like:
```
->leftJoin('sponsor_details', 'sponsors.id', '=', 'sponsor_details.sponsor_id')
->where('sponsor_details.type','=','portal-info')
```

I think it makes it more simple to read and achieves the same thing.